montecarlo/ssa.py

```diff
@@ -99,6 +105,9 @@ def exit(self):
         if self.max_duration is not None:
             if self["time"][-1] >= self.max_duration:
                 return True
+        if self.total_steps is not None:
+            if len(self["time"]) == self.total_steps: 
```

Comment:
My initial thought was to iterate down the `total_steps` parameter to zero, but this is much more clever and contained :)

Reply:
That is how I initially thought and implemented it in my code; a while loop that checked if the length of the "time" dictionary is less than max time_steps. I saw however that you have added the end_time invariant check in the exit method and thought to do the same.